choose-index.md

```diff
@@ -195,197 +195,406 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t1, idx) */ * FROM t1 WHERE JSON_OVERL
+        +---------------------------------+---------+-----------+-----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------+
+        
+        > EXPLAIN SELECT /*+ use_index_merge(t4, mvi1, mvi2) */ * FROM t4 WHERE json_contains(j->'$.a', '[1]') OR json_contains(j->'$.b', '[2]');
+        +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+        | id                            | estRows | task      | access object                                                               | operator info                               |
+        +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+        | IndexMerge_8                  | 19.99   | root      |                                                                             | type: union                                 |
+        | ├─IndexRangeScan_5(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi1(cast(json_extract(`j`, _utf8'$.a') as unsigned array)) | range:[1,1], keep order:false, stats:pseudo |
+        | ├─IndexRangeScan_6(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi2(cast(json_extract(`j`, _utf8'$.b') as unsigned array)) | range:[2,2], keep order:false, stats:pseudo |
+        | └─TableRowIDScan_7(Probe)     | 19.99   | cop[tikv] | table:t4                                                                    | keep order:false, stats:pseudo              |
+        +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+        ```
+
+    - If both `OR` and `AND` appear in the conditions (nested `OR` and `AND`), the conditions that constitute the IndexMerge must all match the semantic of `OR` or the semantic of `AND`, and they cannot make part of them match the semantic of `OR` while the others match the semantic of `AND`. For example:
+        
+        ```sql
+        EXPLAIN SELECT /*+ use_index_merge(t4, mvi1, mvi2) */ * FROM t4 WHERE 1 member of (j->'$.a') AND (2 member of (j->'$.b') OR 3 member of (j->'$.a'));
+        EXPLAIN SELECT /*+ use_index_merge(t4, mvi1, mvi2) */ * FROM t4 WHERE 1 member of (j->'$.a') OR (2 member of (j->'$.b') AND 3 member of (j->'$.a'));
+        ```
+
+        ```sql
+        -- Only two members of (j->'$.b') and three members of (j->'$.a') that match the semantic of OR constitute the IndexMerge. One member of (j->'$.a') that matches the semantic of AND is not included.
```

Comment:
```suggestion
        -- Only the second member of (j->'$.b') and the third member of (j->'$.a') that match the semantic of OR constitute the IndexMerge. the first member of (j->'$.a') that matches the semantic of AND is not included.
```

Reply:
from chatgpt translation？

Reply:
I commented above. They should be `1 member of (j->'$.a')`, `3 member of (j->'$.a')` and `2 member of (j->'$.b')` in the original SQL.
@hfxsd Should we revert to use the original content in the PR?

Reply:
OK let's revert it.